force-app/main/default/classes/ClassesObjectsMethods.cls

```diff
@@ -161,11 +146,16 @@ public with sharing class ClassesObjectsMethods {
      * Example: generateRandomNumber() should return a number like 75.
      * @return A integer between 50 and 100.
      */
-    public static Integer generateRandomNumber() {
-        // Use Math class to generate number between 50 and 100
+    public static Integer generateRandomNumber()
+    {
+        Integer randomNumber = 0; // Initialize random number
+
+        if (randomNumber <= 50 || randomNumber >= 100) // If the number isn't between 50 and 100 including, generate another number
+        {
+            randomNumber = Integer.valueOf(Math.random() * 100);
+        }
```

Comment:
When running this multiple times, it seems to fail. Sometimes the number is under 50
Try to multiply the random number by 50 for the double then cast it into an integer and add 50